dali/python/nvidia/dali/external_source.py

```diff
@@ -124,11 +148,21 @@ def _is_generator_function(x):
     where __call__ is a generator function"""
     if inspect.isgeneratorfunction(x):
         return True
-    if x is None or inspect.isfunction(x):
+    if x is None or inspect.isfunction(x) or inspect.ismethod(x):
         return False
     call = getattr(x, "__call__", None)
     return _is_generator_function(call)
 
+def _accepted_args_count(callable):
```

Comment:
Grammar Nazi attack!
```suggestion
def _accepted_arg_count(callable):
```

Reply:
done